votor/src/certificate_pool.rs

```diff
@@ -387,6 +394,8 @@ impl CertificatePool {
         message: &BLSMessage,
         events: &mut Vec<VotorEvent>,
     ) -> Result<(Option<Slot>, Vec<Arc<CertificateMessage>>), AddVoteError> {
+        // We add stats reporting here because we should almost always have a message.
+        self.stats.maybe_report();
```

Comment:
I'm thinking this might fit better higher up the stack (like in `certificate_pool_ingest` main loop). If we move it up there, we could even make this a sub-component of the `CertificatePoolServiceStats` reporting

Reply:
Hmm I don't think the two stats should be mixed together, they each maintain the status of the corresponding data structure.

But I can move the trigger there. Done.